### PR TITLE
Streamline fixtures, use pytest-xdist, drop Python3.8

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
  - This PR fixes issue #xyz
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)

### What kind of change does this PR introduce?: 

* Sets a version for the `xclim-testdata` repo to prevent breaking changes there from causing issues here.
* Creates two fixtures (open_dataset and get_file) that uses thread safe caching
* Integrates `pytest-xdist` to distribute tests across multiple workers
* Optimizes the testing data setup to reduce potential for data races (with the help of `pooch`).
* Removes a lot of calls to fixtures that weren't needed.
* Updates the `numpy.random.RandomState` call.
* Employs caching of `pip`, `tox`, and testing data to significantly speed up subsequent builds.

### Does this PR introduce a breaking change?: 

Yes. Testing now requires the `pytest-xdist` plugin (has been added to `pyproject.toml` and `environment.yml`). By default, running `pytest` will only use one worker, but `tox` will use up to eight workers.

### Other information: 

https://pytest-xdist.readthedocs.io/en/stable/

### CODE REVIEW

**Review by cehbrecht — APPROVED** [review]

**Comment by cehbrecht** [comment]
@Zeitsperre looks good 👍 ... but probably need some support in adapting PR #342.

**Comment by Zeitsperre** [comment]
There are a handful of tests failing that I can't seem to make sense of. It may be due to breaking changes in more recent versions of `dask` and `xarray`, but it isn't clear to me which changes are the culprit.

Within some of the internals of clisops (e.g. `clisops.ops.base_operation.Operation`) there are direct calls to `roocs-utils`'s `open_xr_dataset`. Is it possible that these helper functions are broken?

One thing for certain is that this affects tests that open paths ending in wildcards (i.e. `path/to/file/*.nc`). I've tried to `load()` these on open, but it doesn't seem to have an effect. I'm totally stumped.

File: tests/conftest.py
Comment (aulemahal) [inline tests/conftest.py:147]: Leaving `seed=None` will make a truly random `rs`,  while passing an int will make a fake random generator. Just saying in case you were trying to remove randomness from the tests.

Comment (Zeitsperre) [reply tests/conftest.py:147]: Good point.

On inspection, this fixture is never called. Perhaps we'll do a cull of the fixtures in another PR.

**Comment by Zeitsperre** [comment]
@cehbrecht Waiting on your review here. Lots of changes, but things are much smoother here.

**Review by cehbrecht — APPROVED** [review]

**Comment by cehbrecht** [comment]
@Zeitsperre thanks for the update. I'm fine with the changes. Moving to `pooch` looks good. Probably we will need some support for the other PR #342 after the merge of this one. Maybe have a release after the merge of this PR? PR #342 also partially replaces some functions of roocs-utils ... but there is more to do.

**Pull request merged**
